File: adlatus-pm-api/src/main/java/de/bitconex/tmf/party/service/OrganizationService.java
package de.bitconex.tmf.party.service;

import de.bitconex.tmf.party.models.Organization;
import de.bitconex.tmf.party.models.OrganizationCreate;
import de.bitconex.tmf.party.models.OrganizationRef;
import de.bitconex.tmf.party.models.OrganizationUpdate;

import java.util.List;
import java.util.Map;

public interface OrganizationService {
    Organization createOrganization(OrganizationCreate organization);

    Organization createOrganization(OrganizationRef organization);

    Boolean deleteOrganization(String id);

    List<Organization> listOrganization(String fields, Integer offset, Integer limit, Map<String, String> allParams);

    Organization patchOrganization(String id, OrganizationUpdate organization);

    Organization retrieveOrganization(String id, String fields);
}

File: adlatus-gui/libs/business/src/lib/party-management/organization.service.ts
import {Inject, Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {
    Organization,
    OrganizationCreate,
    OrganizationUpdate
} from "@adlatus-gui/domain/party-management";

@Injectable({
    providedIn: 'root'
})
export class OrganizationService {

    constructor(private httpClient: HttpClient, @Inject('environment') private environment: Record<string, string | boolean | number>) {
    }

    createOrganization(organizationCreate: OrganizationCreate) {
        return this.httpClient.post<Organization>((<string>this.environment.ORGANIZATION_ENDPOINT), organizationCreate);
    }

    retrieveAllOrganizations() {
        console.log('retrieveAllOrganizations')
        return this.httpClient.get<Organization[]>((<string>this.environment.ORGANIZATION_ENDPOINT));
    }

    retrieveOrganization(id: string) {
        return this.httpClient.get<Organization>(`${(<string>this.environment.ORGANIZATION_ENDPOINT)}/${id}`);
    }

    patchOrganization(id: string, organizationUpdate: OrganizationUpdate) {
        return this.httpClient.patch<Organization>(`${(<string>this.environment.ORGANIZATION_ENDPOINT)}/${id}`, organizationUpdate)
    }

    deleteOrganization(id: string) {
        return this.httpClient.delete<void>(`${(<string>this.environment.ORGANIZATION_ENDPOINT)}/${id}`);
    }
}
